tests/wrapper/test_transactions_methods.py

```diff
@@ -144,13 +145,123 @@ def test_add_transaction_card_not_present(self):
 
-    def test_cancel_transaction(self):
+    def test__capture_or_void_transaction(self):
+        """
+        Dado que existe transação t1
+        Quando for chamado ZoopWrapper().__capture_or_void_transaction(t1.id)
+        Então:
+            - TransactionWrapper.retrieve_transaction deve ter sido chamado
+            - deve ter sido feito um post corretamente
+        """
+        self.set_post_mock(200, CancelTransactionCardFactory(id="foo").to_dict())
+
+        with patch(
+            "zoop_wrapper.wrapper.transaction.TransactionWrapper.retrieve_transaction"
+        ) as mocked_retrieve_transaction:
+
+            t1 = TransactionCreditFactory(id="1", allow_empty=True)
+
+            mocked_retrieve_transaction.return_value = self.build_response_mock(
+                200, instance=t1
+            )
+
+            response = self.client._capture_or_void_transaction(t1.id, "foo")
+            self.assertEqual(response.status_code, 200, msg=response.data)
+            mocked_retrieve_transaction.assert_called_once()
+
+            expected_data = {"amount": t1.amount, "on_behalf_of": t1.on_behalf_of}
+
+            self.mocked_post.assert_called_once_with(
+                f"{self.base_url}/transactions/{t1.id}/foo/",
+                json=expected_data,
+                auth=self.auth,
+            )
+
+    def test__capture_or_void_transaction_amount(self):
+        """
+        Dado que existe transação t1 de 10000
+        Quando for chamado ZoopWrapper().__capture_or_void_transaction(t1.id, 100.00)
+        Então:
+            - TransactionWrapper.retrieve_transaction deve ter sido chamado
+            - deve ter sido feito um post corretamente
+        """
         self.set_post_mock(200, CancelTransactionCardFactory(id="foo").to_dict())
 
         with patch(
             "zoop_wrapper.wrapper.transaction.TransactionWrapper.retrieve_transaction"
         ) as mocked_retrieve_transaction:
 
-            response = self.client.cancel_transaction("foo")
+            t1 = TransactionCreditFactory(id="1", amount=10000, allow_empty=True)
+
+            mocked_retrieve_transaction.return_value = self.build_response_mock(
+                200, instance=t1
+            )
+
+            response = self.client._capture_or_void_transaction(t1.id, "foo", 100.00)
             self.assertEqual(response.status_code, 200, msg=response.data)
             mocked_retrieve_transaction.assert_called_once()
+
+            expected_data = {"amount": 10000, "on_behalf_of": t1.on_behalf_of}
+
+            self.mocked_post.assert_called_once_with(
+                f"{self.base_url}/transactions/{t1.id}/foo/",
+                json=expected_data,
+                auth=self.auth,
+            )
+
+    def test__capture_or_void_transaction_amount_invalid(self):
+        """
+        Dado que existe transação t1 de 10000
+        Quando for chamado ZoopWrapper().__capture_or_void_transaction(t1.id, 100.01)
```

Comment:
O que acontece se o valor for menor? Por exemplo, 99.99.

Reply:
Se for estorno, ocorre um estorno parcial do valor apenas.

Se for captura, ocorre uma captura parcial e o restante é devolvido.